Test failure: `ServiceRequestCancellationTest`

From https://github.com/line/armeria/pull/3539/checks?check_run_id=2563851748#step:7:2310
```
 ServiceRequestCancellationTest > [2] protocol=h2c FAILED
    java.lang.AssertionError: 
    Expecting throwable message:
      <null>
    to contain:
      <"received a RST_STREAM frame: CANCEL">
    but did not.

    Throwable that failed the check:

    com.linecorp.armeria.common.ClosedSessionException
    	at com.linecorp.armeria.common.ClosedSessionException.get(ClosedSessionException.java:36)
    	at com.linecorp.armeria.server.HttpServerHandler.cleanup(HttpServerHandler.java:234)
    	at io.netty.util.concurrent.PromiseTask.runTask(PromiseTask.java:98)
    	at io.netty.util.concurrent.ScheduledFutureTask.run(ScheduledFutureTask.java:170)
    	at io.netty.util.concurrent.AbstractEventExecutor.safeExecute(AbstractEventExecutor.java:164)
    	at io.netty.util.concurrent.SingleThreadEventExecutor.runAllTasks(SingleThreadEventExecutor.java:472)
    	at io.netty.channel.epoll.EpollEventLoop.run(EpollEventLoop.java:384)
    	at io.netty.util.concurrent.SingleThreadEventExecutor$4.run(SingleThreadEventExecutor.java:989)
    	at io.netty.util.internal.ThreadExecutorMap$2.run(ThreadExecutorMap.java:74)
    	at io.netty.util.concurrent.FastThreadLocalRunnable.run(FastThreadLocalRunnable.java:30)
    	at java.base/java.lang.Thread.run(Unknown Source)
        at com.linecorp.armeria.server.ServiceRequestCancellationTest.shouldCompleteLogWhenCancelledByClient(ServiceRequestCancellationTest.java:73)
```

--- a/core/src/test/java/com/linecorp/armeria/server/ServiceRequestCancellationTest.java
+++ b/core/src/test/java/com/linecorp/armeria/server/ServiceRequestCancellationTest.java
@@ -83,5 +83,7 @@ class ServiceRequestCancellationTest {
                     .isInstanceOf(CompletionException.class)
                     .hasCauseInstanceOf(ClosedSessionException.class);
         }
+
+        ctxRef.set(null);
     }
 }

